app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,55 @@
+class InvitationsController < ApplicationController
+  before_action :get_resources, except: [:confirm]
+  skip_before_action :authenticate_user, only: [:confirm]
+
+
+  def new
+    authorize Invitation
+    @invitation = @todo_list.invitations.new
+  end
+
+  def create
+    authorize Invitation
+    @invitation = @todo_list.invitations.new(permitted_attributes(Invitation.new))
+    if @invitation.invited_user_email == current_user.email
```

Comment:
Niestety nie mam tego jeszcze w services.
Szkoleniowo zrobiłam sobie to teraz w modelu: 
```
  validate :not_inviting_self, on: :create
```
```
private
  def not_inviting_self
    errors.add(:invited_user_email, "You can't invite yourself") if self.todo_list.user.email == self.invited_user_email
  end
```